packages/connectivity/src/scp-cf/destination/destination-service.ts

```diff
@@ -212,6 +212,62 @@ export async function fetchCertificate(
   }
 }
 
+/**
+ * Fetch a single destination by name without credentials.
```

Comment:
[req] Could you explain the use case why fetching without tokens is necessary? I remember there was a discussion about this, but not what the reason was.

Reply:
[pp] This function seems quite long to me. Please consider extracting some functions to make it easier to read.

Reply:
You need to know the auth flow to set the `x-tenant` or `x-user-token` headers correctly - see [here](https://api.sap.com/api/SAP_CP_CF_Connectivity_Destination/resource). Thats why we make the first call to get this information and then the second one. This could only be avoided if the user would provide the flow which is not convenient I would say.